tests/components/wyoming/test_conversation.py

```diff
@@ -0,0 +1,239 @@
+        text="success",
+    )
+
+    class TestIntentHandler(intent.IntentHandler):
+        """Test Intent Handler."""
+
+        intent_type = "TestIntent"
+
+        async def async_handle(self, intent_obj: intent.Intent):
+            """Handle the intent."""
+            assert intent_obj.slots.get("entity", {}).get("value") == "value"
+            return intent_obj.create_response()
+
+    intent.async_register(hass, TestIntentHandler())
+
+    with patch(
+        "homeassistant.components.wyoming.conversation.AsyncTcpClient",
+        MockAsyncTcpClient([test_intent.event()]),
+    ):
+        result = await agent.async_process(
+            conversation.ConversationInput(
+                "test text",
+                Context(),
+                conversation_id=conversation_id,
+                device_id=None,
+                language=hass.config.language,
+            )
+        )
+
+    assert result.response.response_type == intent.IntentResponseType.ACTION_DONE
+    assert result.response.speech, "No speech"
+    assert result.response.speech.get("plain", {}).get("speech") == "success"
+    assert result.conversation_id == conversation_id
+
+
+async def test_intent_handle_error(
+    hass: HomeAssistant, init_wyoming_intent: ConfigEntry
+) -> None:
+    """Test error during handling when an intent is recognized."""
+    agent = conversation.async_get_agent(hass, init_wyoming_intent.entry_id)
+    assert agent is not None
+
+    test_intent = Intent(name="TestIntent", entities=[], text="success")
+
+    class TestIntentHandler(intent.IntentHandler):
+        """Test Intent Handler."""
+
+        intent_type = "TestIntent"
+
+        async def async_handle(self, intent_obj: intent.Intent):
+            """Handle the intent."""
+            raise intent.IntentError
+
+    intent.async_register(hass, TestIntentHandler())
+
+    with patch(
+        "homeassistant.components.wyoming.conversation.AsyncTcpClient",
+        MockAsyncTcpClient([test_intent.event()]),
+    ):
+        result = await agent.async_process(
+            conversation.ConversationInput(
+                "test text",
+                Context(),
+                conversation_id=None,
+                device_id=None,
+                language=hass.config.language,
+            )
+        )
+
+    assert result.response.response_type == intent.IntentResponseType.ERROR
+    assert result.response.error_code == intent.IntentResponseErrorCode.FAILED_TO_HANDLE
+
+
+async def test_not_recognized(
+    hass: HomeAssistant, init_wyoming_intent: ConfigEntry
+) -> None:
+    """Test when an intent is not recognized."""
+    agent = conversation.async_get_agent(hass, init_wyoming_intent.entry_id)
+    assert agent is not None
+
+    with patch(
+        "homeassistant.components.wyoming.conversation.AsyncTcpClient",
+        MockAsyncTcpClient([NotRecognized(text="failure").event()]),
+    ):
+        result = await agent.async_process(
```

Comment:
We shuldn't talk directly to the agent. Instead call `conversation.async_converse`

Reply:
Otherwise you are not testing that the method is actually called correctly from the conversation agent.